mvcc/kvstore.go

```diff
@@ -160,6 +166,56 @@ func (s *store) Hash() (hash uint32, revision int64, err error) {
 	return h, s.currentRev, err
 }
 
+func (s *store) HashByRev(rev int64) (hash uint32, currentRev int64, compactRev int64, err error) {
+	readTx := s.b.ReadTx()
+	for {
```

Comment:
one more thought. if I want to hash all the ways between [compaction rev, current rev], I can't simply just hash all the key that has rev > compaction rev and <= current rev because compaction deletes historic revisions for a key if it is not tombstone. In that case, I need to skip key that's historic and has rev < compaction rev. hence, I need a way to determine if a key is historic. How would that be achieve easily just by looping through kv space?

It seems to me that waiting for the compaction to finish can be a simpler approach where i don't have to worry about checking tombstone and historic keys that has rev < compaction rev.